src/main/java/org/reaktivity/nukleus/kafka/internal/stream/KafkaCacheClientProduceFactory.java

```diff
@@ -615,6 +616,12 @@ private void onClientInitialData(
                 flushClientFanInitialIfNecessary(traceId);
             }
 
+            if ((flags & FLAGS_INCOMPLETE) != 0x00)
+            {
+                markEntryDirty(stream.partitionOffset);
+                flushClientFanInitialIfNecessary(traceId);
```

Comment:
No need to `flush` given that entry has been marked dirty. The message does not need to be produced.